src/request-queue.js

```diff
@@ -103,8 +103,13 @@ export class RequestQueue {
             }
 
             // :TRICKY: Metrics-only code
-            if (options && requests[key].options && requests[key].options._metrics) {
-                requests[key].options._metrics['rrweb_full_snapshot'] ||= options._metrics['rrweb_full_snapshot']
+            if (
+                options &&
+                requests[key].options &&
+                requests[key].options._metrics &&
+                typeof requests[key].options._metrics['rrweb_full_snapshot'] === 'undefined'
+            ) {
+                requests[key].options._metrics['rrweb_full_snapshot'] = options._metrics['rrweb_full_snapshot']
```

Comment:
This changes semantics in a way that makes this metric useless.

The goal is to measure if any of the snapshots sent is a FULL_SNAPSHOT. After this change this returns only true if the first snapshot is FULL_SNAPSHOT.

Reply:
Damn, thanks for the context! Will this change fix it?

Reply:
Yes, though the proper way to extend 'a ||= b` is `a = (a || b)`, but changing the if gives the same result.